lib/player.js

```diff
@@ -703,6 +721,77 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     if (mediaElement) {
       this.attach(mediaElement, /* initializeMediaSource= */ true);
     }
+
+    if (canvas) {
+      this.canvas_ = canvas;
+    }
+  }
+
+  /**
+   * Create a shaka.lcevc.Dil object
+   * @private
+   */
+  createDIL_() {
+    if (this.lcevcDil_ == null) {
+      this.lcevcDil_ = new shaka.lcevc.Dil(
+          this.video_,
+          this.canvas_,
+          this.lcevcDilConfig_,
+      );
+      this.canvas_ = this.lcevcDil_.canvas_;
+      this.mediaSourceEngine_.updateLcevcDil(this.lcevcDil_);
+    }
+  }
+
+  /**
+   * Close a shaka.lcevc.Dil object
+   * @private
+   */
+  closeDIL_() {
+    if (this.lcevcDil_ != null) {
+      if (this.lcevcDil_.media_) {
+        this.lcevcDil_.media_.style.display = 'block';
+        this.lcevcDil_.canvas_.style.display = 'none';
+      }
+      this.lcevcDil_.close();
+      delete this.lcevcDil_;
```

Comment:
Don't both delete and set to null.  Setting to null should suffice for GC, and we generally don't delete fields.

Reply:
This is rectified.